tests/test_architecture.py

```diff
@@ -11,6 +11,22 @@
 GNAME = "framework-testing"
 
 
+def summary_exception(values: np.ndarray,
+                      raise_exception: bool = False) -> int:
+    if raise_exception:
+        raise ValueError("Summary exception raised.")
+
+    return len(values)
+
+
+def summary_memory_error(values: np.ndarray,
```

Comment:
Since you are maintaining generic methods for testing in MFETestClass, it would also be interesting to put this function in MFETestClass.
is there a constraint to don't put this function in MFETestClass?